src/api/app/controllers/webui/package_controller.rb

```diff
@@ -984,18 +984,40 @@ def meta
   end
 
   def save_meta
-    begin
-      frontend.put_file(params[:meta], :project => @project, :package => @package, :filename => '_meta')
-    rescue ActiveXML::Transport::Error => e
-      message = e.summary
-      flash[:error] = message
-      @meta = params[:meta]
-      render :text => message, :status => 400, :content_type => 'text/plain'
+    meta_xml = Xmlhash.parse(params[:meta])
```

Comment:
Ok! But I guess you can use 
``
validate('package', meta_xml)
``
to validate if the XML is valid! But you have to catch the exception then!